### PR TITLE
fix a bunch of clippy warnings

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -137,10 +137,8 @@ fn run_parallel(args: &Arc<Args>) -> Result<u64> {
             Continue
         })
     });
-    if !args.paths().is_empty() && paths_searched.load(Ordering::SeqCst) == 0 {
-        if !args.no_messages() {
-            eprint_nothing_searched();
-        }
+    if !args.paths().is_empty() && paths_searched.load(Ordering::SeqCst) == 0 && !args.no_messages() {
```
Comment (BurntSushi) [inline src/main.rs]: Please drop this entire commit. I prefer the existing style.

Comment (matthiaskrgr) [reply src/main.rs]: done :)

**Comment by matthiaskrgr** [comment]
I added another commit that adds clippy as optional dependency.
```` cargo build --features clippy```` will compile with static analysis :)

File: Cargo.toml [hunk 2]
```diff
@@ -33,6 +33,7 @@ path = "tests/tests.rs"
 atty = "0.2.2"
 bytecount = "0.1.4"
 clap = "2.26"
+clippy = {version = "*", optional = true}
```
Comment (BurntSushi) [inline Cargo.toml]: Sorry, but please remove this and any corresponding `cfg`s. I don't want to add a very unstable linter to ripgrep's dependencies.

Comment (matthiaskrgr) [reply Cargo.toml]: This is not a default dependency, clippy is *only* pulled in if we explicitly build with "--features clippy"

Comment (BurntSushi) [reply Cargo.toml]: Yes, I understand. I don't want it.

Comment (matthiaskrgr) [reply Cargo.toml]: Alright, I removed it.

**Pull request merged**